### ISSUE TITLE
terraform executable file not found in $PATH, but it is there... in the $PATH

### ISSUE THREAD

**Issue by boldandbusted** [gruntwork-io/terragrunt#268]
```shell
$ terragrunt init
[terragrunt] [REDACTED] 2017/08/22 19:06:42 Running command: terraform --version
[terragrunt] 2017/08/22 19:06:42 exec: "terraform": executable file not found in $PATH
[terragrunt] 2017/08/22 19:06:42 Unable to determine underlying exit code, so Terragrunt will exit with error code 1
$ 
```

but...

```shell
$ which terraform
/home/jesse/bin/terraform
$ 
```

and...

```shell
$ terraform --version
Terraform v0.10.2

$
```

How have I sinned?

**Comment by brikis98** [comment]
Did you install Terraform or Terragrunt using snap by any chance?

**Comment by boldandbusted** [comment]
@brikis98 Sorry for the delay - didn't see you replied. Nope, direct downloads and unzips to ~/bin, for both terraform and terragrunt.

**Comment by boldandbusted** [comment]
Wow, I fixed it, but I'm still mystified. If you have ~/ in your PATH, at least in Bash 4.3.48(1)-release, this problem manifests.

With 'terragrunt' and 'terraform' in /home/jesse/bin/, I had:
export PATH="~/bin:~/.local/bin/:$PATH"

Then, I changed that line to:
export PATH="/home/jesse/bin:/home/jesse/.local/bin/:$PATH"

And the problem disappeared.

Apparently I violated a basic law of the shell, but was only put in limbo, without an error reported? Very strange that other binaries and shell scripts worked in this mode... for years. Leaving open for you to close, as perhaps this affected your 'snap' users?
